crates/index/src/lib.rs

```diff
@@ -538,7 +564,20 @@ impl<O: Op> IndexProtect<O> {
         self.startup.set(IndexStartup {
             sealeds: startup_sealeds,
             growings: startup_growings,
+            flexible: self.flexible_get(),
         });
         swap.swap(view);
     }
+    fn flexible_set(&mut self, flexible: IndexFlexibleOptions) {
+        let src = self.startup.get();
+        self.startup.set(IndexStartup {
+            sealeds: src.sealeds.clone(),
+            growings: src.sealeds.clone(),
+            flexible,
+        });
+    }
+    fn flexible_get(&self) -> IndexFlexibleOptions {
+        let src = self.startup.get();
+        src.flexible.clone()
+    }
```

Comment:
These functions are not necessary.

We should always get flexible options in a `IndexView`, not a `IndexProtect`.